plugins/feedback-backend/src/api/jiraApiService.ts

```diff
@@ -1,90 +1,105 @@
 import axios from 'axios';
 
-export const createJiraTicket = async (options: {
-  host: string;
-  authToken: string;
-  projectKey: string;
-  summary: string;
-  description: string;
-  tag: string;
-  feedbackType: string;
-  reporter?: string;
-}): Promise<any> => {
-  const {
-    host,
-    authToken,
-    projectKey,
-    summary,
-    description,
-    tag,
-    feedbackType,
-    reporter,
-  } = options;
-  const requestBody = {
-    fields: {
-      ...(reporter && {
-        reporter: {
-          name: reporter,
+/**
+ * @param host
+ * @param authToken
+ * @param hostType
+ */
+export class JiraApiService {
+  private readonly hostType: string;
+  constructor(
+    private host: string,
+    private authToken: string,
+    hostType?: string,
+  ) {
+    if (!hostType) this.hostType = 'SERVER';
+    else this.hostType = 'CLOUD';
```

Comment:
This statement makes no sense, as it will use `CLOUD` even if the user specifies `hostType = 'SERVER'`due to the truthy check used.

Since the hostType is a string, why not just assign that to the `this.hostType` property?

```suggestion
    this.hostType = hostType ?? 'SERVER';
```

Or you can do the same in the contructor parameters by providing a default value:
```ts
  constructor(
    // ...
    private hostType: string = 'SERVER',
  ) {
```

Reply:
Done